src/tikzpy/tikz_environments/tikz_picture.py

```diff
@@ -201,7 +208,16 @@ def show(self, quiet: bool = False) -> None:
         or open the PDF in the user's browser.
         """
         pdf_file = self.compile(quiet=quiet)
-        webbrowser.open_new(str(pdf_file.as_uri()))
+        if JUPYTER_ENABLED:
+            page_pngs = convert_from_path(pdf_file, transparent=True)
+            for img in page_pngs:
+                display(img)
+        else:
+            webbrowser.open_new(str(pdf_file.as_uri()))
+
+    def close(self) -> None:
+        if self.TEMP_DIR is not None and self.TEMP_DIR.is_dir():
+            rmtree(self.TEMP_DIR)
```

Comment:
One thing to avoid in the library is any code that removes files from the user's computer. It's better to delegate that to the user, because the library could in theory accidentally delete an important file off the user's computer (and even if it's the user's fault for not reading the docs and engineering that situation, it's something to avoid)

Reply:
Agree! However I'm not familiar with `tempfile`. Could it hold the temp dir until the final output is rendered?

```
tempfile.mkdtemp(dir=os.getcwd())
```
It is no problem to get a temp dir in the current directory (not placing it in user directory).